### PR TITLE
[commission] add joiner advertisement to the discovery request handler

### PR DESCRIPTION
**Pull request by gjc13**

### CODE REVIEW

File: include/openthread/thread.h [hunk 1]
```diff
@@ -811,9 +811,12 @@ void otThreadRegisterParentResponseCallback(otInstance *                   aInst
...
 {
-    otExtAddress mExtAddress;   ///< IEEE 802.15.4 Extended Address of the requester
-    uint8_t      mVersion : 4;  ///< Thread version.
-    bool         mIsJoiner : 1; ///< Whether is from joiner.
+    otExtAddress mExtAddress;                            ///< IEEE 802.15.4 Extended Address of the requester
+    uint8_t      mVersion : 4;                           ///< Thread version.
+    bool         mIsJoiner : 1;                          ///< Whether is from joiner.
+    uint32_t     mOui;                                   ///< The vendor Oui
+    uint8_t      mAdvDataLength;                         ///< The advertisement data length
+    uint8_t      mAdvData[OT_JOINER_ADVDATA_MAX_LENGTH]; ///< The advertisement data length
```
[the remark this answers is not in the document]
Comment (gjc13) [reply include/openthread/thread.h:819]: I copy data here to be safe. If it's a pointer it's quite likely to point to somewhere inside an otMessage and we can't be very sure how user will handle this pointer (what if they store it).

[the remark this answers is not in the document]
Comment (wgtdkp) [reply include/openthread/thread.h:819]: I agree with @gjc13 that we should copy here.

File: include/openthread/thread.h [hunk 2]
```diff
@@ -811,9 +811,12 @@ void otThreadRegisterParentResponseCallback(otInstance *                   aInst
  */
 typedef struct otThreadDiscoveryRequestInfo
 {
-    otExtAddress mExtAddress;   ///< IEEE 802.15.4 Extended Address of the requester
-    uint8_t      mVersion : 4;  ///< Thread version.
-    bool         mIsJoiner : 1; ///< Whether is from joiner.
+    otExtAddress mExtAddress;                            ///< IEEE 802.15.4 Extended Address of the requester
+    uint8_t      mVersion : 4;                           ///< Thread version.
+    bool         mIsJoiner : 1;                          ///< Whether is from joiner.
+    uint32_t     mOui;                                   ///< The vendor Oui
```
Comment (abtink) [inline include/openthread/thread.h:817]: I am fine with the PR and this approach (for first implementation)

Just want to bring up a topic related to this (to consider for future)

We may have more than one Joiner Adv info (for different OUIs) in the same discovery message. How to handle this situation?
- One way is to make this a linked-list or array of "Joiner Adv Info".
- Second way is to invoke the callback multiple times (for same received discovery msg) with different `otThreadDiscoveryRequestInfo` entries (each time for a different OUI) - I would prefer this.

**Comment by jwhui** [comment]
Closing stale PR.

**Pull request closed without merging**